### DISCUSSION TITLE
How to pixelate/ hide certain regions of a camera stream?

### DISCUSSION THREAD

**Discussion by hydroid7** [roflcoopter/viseron#715] [Q&A] [answered]
Hi,
is there a possibility to pixelate or hide certain regions of a stream so that it does not get recorded?
Can it maybe done with some FFMPEG filters?

Thank you in advance!

**Comment by roflcoopter** [comment]
Sorry for the late reply, missed this question.
Should be possible with some FFmpeg filters yes.

Something like this came up when doing a quick google search:
```bash
ffmpeg -i input -filter:v "crop=out_w:out_h:x:y" output
```
Replace `out_w` and `out_h` with the width and height of the output video, and `x` and `y` with the top-left corner coordinates of the cropping area.

You can use the config option `video_filters` to apply it to the camera.

Dont have time right now to test it out myself, but please let me know how it works out for you!

**Comment by hydroid7** [comment, marked as answer]
Thank you for the suggestion, I've tried it with crop, but then I get strange artifacts on my stream. I assume it is because of the changed resolution.
Instead you can use drawbox:
```yaml
video_filters:
  - drawbox=x=0:y=0:w=1920:h=200:color=black@1:t=fill
  - drawbox=x=50:y=200:w=800:h=600:color=black@1:t=fill
```
and it works well. You can even apply it multiple times to black out certain non-rectangular areas.

**Reply by roflcoopter** [reply]
Nice! So drawbox is working well? If so I will add it to the docs as an example

**Reply by hydroid7** [reply]
Its working really good, as I mentioned before, you can even specify multiple filters and they are easy to set up!
